### PR TITLE
add ttl and qps policy for scale up and down

### PR DESCRIPTION
**Pull request by tx19980520**

### CODE REVIEW

File: pkg/event/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+			target = ttl.Target
+		} else {
+			target = qps.Target
+		}
+		trend = source.Increase
+	} else {
+		// qps:Decrease, always take ttl?
+		trend = source.Decrease
+		target = ttl.Target
+	}
```
Comment (linxuyalun) [inline pkg/event/metrics/metrics.go:105]: 1. qps: increase/decrease, ttl: increase. --- impossible, ttl always do decrease action;
2. qps: increase, ttl: decrease --- apply the higher target one;
3. qps: decrease, ttl: decrease --- always apply ttl, qps only report the status

so I think the code is reasonable

Comment (linxuyalun) [reply pkg/event/metrics/metrics.go:105]: maybe use `qps.Trend == source.Increase` and `qps.Trend == source.Dncrease` is more clear than `if else` ?

**Review by linxuyalun — COMMENTED** [review]
See conversation

File: cmd/root.go [hunk 2]
```diff
@@ -83,6 +89,10 @@ func basicFlagInit() {
 	viper.BindPFlag(env.Mock, rootCmd.Flags().Lookup(env.Mock))
 	rootCmd.Flags().BoolP(env.StaticMiddleware, "i", false, "whether to use StaticMiddleware")
 	viper.BindPFlag(env.StaticMiddleware, rootCmd.Flags().Lookup(env.StaticMiddleware))
+	rootCmd.Flags().BoolP(env.QPSMiddleware, "q", false, "whether to use QPSMiddleware")
+	viper.BindPFlag(env.QPSMiddleware, rootCmd.Flags().Lookup(env.QPSMiddleware))
+	rootCmd.Flags().DurationP(env.TTL, "T", 20*time.Second, "set process default ttl")
+	viper.BindPFlag(env.TTL, rootCmd.Flags().Lookup(env.TTL))
 }
 
 func policyFlagInit() {
```
[the remark this answers is not in the document]
Comment (tx19980520) [reply cmd/root.go:98]: see `pkg/runner/lsds`

File: pkg/event/metrics/metrics.go (hunk 1 above)
Comment (tx19980520) [reply pkg/event/metrics/metrics.go:105]: I add a `else if qps.Trend == source.Decrease`

**Pull request merged**